### PR TITLE
Eliminate use of fgets() and fixed sized buffers

### PR DESCRIPTION
**Pull request by bje-**
Let's use C++ so we don't have to use fixed sized buffers with all of their associated problems!

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -2629,12 +2629,14 @@ int SamApp::RevisionNumber()
 #else
 	wxString path = SamApp::GetRuntimePath() + "/patches/patch_lnx.txt";
 #endif
-	if ( FILE *fp = fopen( (const char*)path.c_str(), "r" ) )
+	std::ifstream fp;
```
Comment (sjanzou) [inline src/main.cpp:2632]: Changes would have to be tested with Version Upgrade script and with web updated

File: src/widgets.cpp [hunk 2]
```diff
@@ -2310,8 +2312,9 @@ class AFDataLifetimeArrayDialog : public wxDialog
 		{
 			wxFileDialog dlg(this, "Select data file to import");
 			if (dlg.ShowModal() != wxID_OK) return;
-			FILE *fp = fopen(dlg.GetPath().c_str(), "r");
-			if (!fp)
+			std::ifstream fp;
```
Comment (sjanzou) [inline src/widgets.cpp:2315]: Tested by @dguittet - should be fine

File: src/invoke.cpp [hunk 3]
```diff
@@ -3447,19 +3448,20 @@ static void fcall_sam_async( lk::invoke_t &cxt )
 	cxt.result().empty_hash();
 
 	lk_string fn = cxt.arg(0).as_string();
-	FILE *fp = fopen(fn.c_str(), "r");
-	if (!fp)
+	std::ifstream fp;
```
Comment (sjanzou) [inline src/invoke.cpp:3451]: Would have to be tested for any file access issues using the PVRPM script that uses these functions

File: src/invoke.cpp [hunk 4]
```diff
@@ -4007,18 +4010,19 @@ void lhs_threaded(lk::invoke_t &cxt, wxString &workdir, int &sv, int &num_sample
 	if (wxFileExists(workdir + "/LHS.ERR"))
 	{
 		err_msg = "LHS error.  There could be a problem with the input setup.";
-		FILE *ferr = fopen(wxString(workdir + "/LHS.ERR").c_str(), "r");
-		if (ferr)
+		std::ifstream ferr;
```
Comment (sjanzou) [inline src/invoke.cpp:4013]: Changes would have to be tested with LHS sampling in PVRPM code for any issue with file access

**Comment by dguittet** [comment]
@sjanzou Thanks for looking through the changes. Do you or @janinefreeman have PVRPM code handy to run?

File: src/invoke.cpp (hunk 4 above)
Comment (bje-) [reply src/invoke.cpp:4013]: There really should be a testsuite!

Comment (janinefreeman) [reply src/invoke.cpp:4013]: We agree! We're still working on catching up on our test coverage for our codebases- it's easier to add tests to sections we're actively working on, but we hope to be able to get more complete coverage through the upcoming years! If you happen to get motivated to add some tests, we'd be thrilled! =)

Comment (sjanzou) [reply src/invoke.cpp:4013]: @bje- Could you please build SAM from this branch and run a stochastic simulation? (Right below the "Simulate" button on lower left-hand side of SAM case page).

File: src/widgets.cpp (hunk 2 above)
Comment (bje-) [reply src/widgets.cpp:2315]: If this PR can be approved and merged, I can do back and fix all remaining instances of `FILE *` uses.

File: src/invoke.cpp (hunk 4 above)
Comment (dguittet) [reply src/invoke.cpp:4013]: Just did this and the stochastic simulation worked! Can we merge this?

Comment (janinefreeman) [reply src/invoke.cpp:4013]: @sjanzou thoughts?

**Comment by bje-** [comment]
If there are any remaining concerns about this change, it's possible to (more or less) check it by inspection.

**Pull request closed without merging**